### ISSUE TITLE
Setting table_name breaks primary_key on Rails 5

### ISSUE THREAD

**Issue by coryondemand** [rails/rails#25975]
### Steps to reproduce

Create new rails application (I'm using postgres which may or may not be relevant).
Add database.yml info.

Add a single model.

Set self.table_name to something. In my case my model is called Item and table_name = :items_with_state. items_with_state is a view in postgres.

In the rails console ask the model for the primary_key ie. Item.primary_key

nil is returned

If I explicitly set primary_key on the model then the proper result is returned.

### Expected behavior
The primary key should be returned.

### Actual behavior
nil was returned

### System configuration
**Rails version**:
Rails 5.0.0

**Ruby version**:
ruby 2.3.1p112 (2016-04-26 revision 54768) [x86_64-darwin15]

**Comment by rafaelfranca** [comment]
Could you create a reproduction script with this template https://github.com/rails/rails/blob/master/guides/bug_report_templates/active_record_master.rb?

**Comment by cloke** [comment]
After doing some more investigation and trying to reproduce using the above script we were able to narrow the issue down to a very specific use case, but something that is hard to show in a simple script (still working on that).

The issue seems to be with using table_name where the table in postgres is a view. So for example:

We have a view in postgres called item_with_state

app.models/item_with_state.rb
class ItemWithState < ActiveRecord::Base
end
Rails console: "ItemWithState.primary_key" returns "id"

class Item < ActiveRecord::Base
  self.table_name = :item_with_state
end
Rails console: "Item.primary_key" returns "nil"

So to sum up any model that uses table_name and the table_name is a view it will return nil for primary_key.

**Comment by kalrover** [comment]
I think the ActiveRecord returns `nil` primary_key is an expected behaviour due to postgresql can't have primary keys.

@cloke 
The reason why you got`"id"` when not explicitly set the table_name is by default ActiveRecord translates the class name to plural form, so that its unable to find the translated plural table name in the database. As a result, the `primary_key` returns the `"id"` string. Please refer this:
https://github.com/rails/rails/blob/master/activerecord/lib/active_record/attribute_methods/primary_key.rb#L97-L102
```ruby
            if ActiveRecord::Base != self && table_exists?
              pk = connection.schema_cache.primary_keys(table_name)
              suppress_composite_primary_key(pk)
            else
              'id'
            end
```
I think when you set `ActiveRecord::Base.pluralize_table_names = false`, the `primary_key` would return `nil`.

**Comment by pixeltrix** [comment]
@cloke this is a duplicate of #16555 - since we don't know what the primary key is on a view you need to specify it:

``` ruby
class Item < ActiveRecord::Base
  self.table_name = :item_with_state
  self.primary_key = :id
end
```

Because of fde0d02 we don't assume that the primary key is `id`.
